Initial pass removing old linting documentation (#6193)
microsoft/vscode-docs: docs/python/linting.md

## Choose a linter

Install the linting tool of your choice from the marketplace. We support the following linting extensions:

| Linter | Extension                                                                       |
| ------ | ------------------------------------------------------------------------------- |
| Pylint | https://marketplace.visualstudio.com/items?itemName=ms-python.pylint            |
| flake8 | https://marketplace.visualstudio.com/items?itemName=ms-python.flake8            |
| mypy   | https://marketplace.visualstudio.com/items?itemName=ms-python.mypy-type-checker |

Linting extensions offered by the community:

| Linter | Extension                                                              |
| ------ | ---------------------------------------------------------------------- |
| Ruff   | https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff |
| mypy   | https://marketplace.visualstudio.com/items?itemName=matangover.mypy    |

> > **Note**: If you don't find your preferred linter in the table above or in the marketplace, you can add support via an extension. The [Python Extension Template](/api/advanced-topics/python-extension-template.md) makes it easy to integrate new Python tools into VS Code.

Reviewer: ```suggestion
> **Note**: If you don't find your preferred linter in the table above or in the marketplace, you can add support for it via an extension. You can use the [Python Extension Template](/api/advanced-topics/python-extension-template.md) to integrate new Python tools into VS Code.
```

## General Settings

| Setting        | Default      | Description                                                                                                                                                                                            |
| -------------- | ------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | args           | `[]`         | Will be passed to the linter. **Note**: The officially supported linters run on individual opened file. Make sure your configuration applies in that scenario.                                         |

Reviewer: ```suggestion
| args           | `[]`         | Arguments to be passed to the linter. **Note**: The officially supported linters run on individual open files. Make sure your configuration applies in that scenario. |
```

> | importStrategy | `useBundled` | `useBundled` means extension uses bundled version of the linter. `fromEnvironment` means it will attempt to load from your selected python environment first, otherwise falls back to bundled version. |

Reviewer: ```suggestion
| importStrategy | `useBundled` | When set to `useBundled`, the extension will use the version of the tool that it ships with. When set to `fromEnvironment`, it will attempt to load from your selected Python environment first, otherwise it will fall back to the bundled version. |
```

## Disable linting

Linters, if installed, are enabled by default. You can disable them by [disabling the extension](/docs/editor/extension-marketplace.md#disable-an-extension) per workspace.

## Run linting

Linting will automatically run when a Python file is opened or saved.

Errors and warnings are shown in the **Problems** panel for open files, and are also highlighted in the code editor. Hovering over an underlined issue displays the details:

![Linting messages in the editor and the Problems panel](images/linting/lint-messages.png)

## Code Actions

> Some linters may offer code actions that can help address reported problems.

Reviewer: ```suggestion
Some linters may offer code actions that can help address reported problems. You can refer to the [Feature Contributions](/docs/editor/extension-marketplace#extension-details) section under your preferred linter extension to find out what code actions it offers. 
```

## Logging

Logs for linters are available in the `Output` panel when you select `<linter name>` from the drop down menu.

> You can change the log level for a linter extension by running the command pallet command `Developer: Set Log Level` select the extension from the extension logs group. and change the level.

Reviewer: ```suggestion
You can change the log level for a linter extension by running the **Developer: Set Log Level** command from the Command Palette (`kb(workbench.action.showCommands)`). Select the extension from the Extension Logs group, and then select the desired log level. 
```

## Severity

> Linters report issues with some predefined severity. This can be changed using `severity` setting for the linter. Refer to readme for the linter extension of your choice for more details on the supported values and severity levels.

Reviewer: ```suggestion
Linters report issues with some predefined severity. This can be changed using `severity` setting for the linter. Refer to each linter extension's README for more details on the supported values and severity levels.
```

## Troubleshooting linting

| Issue | Cause | Solution |
| --- | --- | --- |

> | No problems reported | No python selected | Look at the logs for the linter you are using. It should say which python is being used. If there is no python selected, there should be some instructions on how to select an interpreter. |

Reviewer: ```suggestion
| No problems reported | No Python has been selected for your workspace. | Look at the logs for the linter you are using and check the path to the Python environment it's using. If there is no Python selected, run the **Python: Select Interpreter** command from the Command Palette and select an existing interpreter for your workspace.  |
```

> **Note**: Stylistic and syntactical code detection is enabled by the Language Server. To enable third-party linters for additional problem detection, you can enable them by using the **Python: Select Linter** command and selecting the appropriate linter.

> ## Getting a linter

Reviewer 2: ```suggestion
## Choosing a linter
```

> ## Choosing a linter

Reviewer 2: ```suggestion
## Choose a linter
```

> You can install the linting tool of your choice from the marketplace. We support following linting extensions:

Reviewer 2: ```suggestion
Install the linting tool of your choice from the marketplace. We support the following linting extensions:
```

> You can install linting tools from the marketplace. Look for the linter you want to use. Install the extension for it. We support following linting extensions:

Reviewer 2: ```suggestion
You can install the linting tool of your choice from the marketplace. We support following linting extensions:
```

| Linter | Extension |
| --- | --- |
| Pylint | https://marketplace.visualstudio.com/items?itemName=ms-python.pylint |
| flake8 | https://marketplace.visualstudio.com/items?itemName=ms-python.flake8 |
| mypy | https://marketplace.visualstudio.com/items?itemName=ms-python.mypy-type-checker |

> Offered by community:

Reviewer 2: ```suggestion
Linting extensions offered by the community:
```

| Linter | Extension |
| --- | --- |
| Ruff | https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff |
| mypy | https://marketplace.visualstudio.com/items?itemName=matangover.mypy |

> > **Note**: If you don't find your preferred linter in the table above or in the market place, you can add support via an extension. The [Python Extension Template](/api/advanced-topics/python-extension-template.md) makes it easy to integrate new Python tools into VS Code.

Reviewer 2: ```suggestion
> **Note**: If you don't find your preferred linter in the table above or in the marketplace, you can add support via an extension. The [Python Extension Template](/api/advanced-topics/python-extension-template.md) makes it easy to integrate new Python tools into VS Code.
```

## General Settings

| Setting | Default | Description |
| --- | --- | --- |
| args | `[]` | Will be passed to the linter. **Note**: The officially supported linters run on individual opened file. Make sure your configuration applies in that scenario. |
| importStrategy | `useBundled` | `useBundled` means extension uses bundled version of the linter. `fromEnvironment` means it will attempt to load from your selected python environment first, otherwise falls back to bundled version. |

> ## Disabling Linter

Reviewer 2: ```suggestion
## Disable linting
```

Reviewer 2: Consistency with the "Run Linting" header

> Linters, if installed, are enabled by default. You can disable them by disabling the extension per workspace.

Reviewer: ```suggestion
Linters, if installed, are enabled by default. You can disable them by [disabling the extension](/docs/editor/extension-marketplace.md#disable-an-extension) per workspace.
```

> Linters if installed are enabled by default. You can disable them by disabling the extension per workspace.

Reviewer 2: ```suggestion
Linters, if installed, are enabled by default. You can disable them by disabling the extension per workspace.
```

## Run linting

> Linting will automatically run on any file on Open or Save. It will clear out the problems for a file on Close.

Reviewer: ```suggestion
Linting will automatically run when a Python file is opened or saved. Errors and warnings are reported in the Problems window while the file is open, and they will disappear when the file is closed. 
```

> Linter will automatically run on any file on Open or Save. It will clear out the problems for a file on Close.

Reviewer 2: ```suggestion
Linting will automatically run on any file on Open or Save. It will clear out the problems for a file on Close.
```

> Issues are shown in the **Problems** panel and as wavy underlines in the code editor. Hovering over an underlined issue displays the details:

Reviewer: ```suggestion
Errors and warnings are shown in the **Problems** panel for open files, and are also highlighted in the code editor. Hovering over an underlined issue displays the details:
```

![Linting messages in the editor and the Problems panel](images/linting/lint-messages.png)

## Code Actions

> Some linters may offer code actions on problems, which can halp address the issue.

Reviewer: ```suggestion
Some linters may offer code actions that can help address reported problems.
```

## Logging

> Logs for linters should be available in the `Output` tab. Select `<linter name>` from the drop down.

Reviewer: ```suggestion
Logs for linters are available in the `Output` panel when you select `<linter name>` from the drop down menu.
```

> Logs for linter should be available in the `Output` tab. Select `<linter name>` from the drop down.

Reviewer 2: ```suggestion
Logs for linters should be available in the `Output` tab. Select `<linter name>` from the drop down.
```

You can change the log level for a linter extension by running the command pallet commmand `Developer: Set Log Level` select the extension from the entension logs group. and change the level.

> ## Troubleshooting linting

Reviewer: Should we remove this troubleshooting section? It feels heavily applied to the built-in functionality

Author: This was one of the items I wanted to discuss. I think we need to add items here that belong to the extension cases.

Author: "LanguageServer crashes 5 times" this case should not happen any more. I have updated the Trouble shooting with the most common issue I see with linting.
